include/igl/predicates/ear_clipping.cpp

```diff
@@ -13,42 +20,48 @@ IGL_INLINE void igl::predicates::ear_clipping(
   Eigen::PlainObjectBase<DerivedF>& eF,
   Eigen::PlainObjectBase<DerivedP>& nP
 ){
+  typedef typename DerivedF::Scalar Index;
+  typedef typename DerivedP::Scalar Scalar;
+  static_assert(std::is_same<typename DerivedI::Scalar,
+                             typename DerivedF::Scalar>::value,
+                "index type should be consistent");
   
   // check whether vertex i is an ear
   auto is_ear = [](
     const Eigen::MatrixBase<DerivedP>& P,
     const Eigen::MatrixBase<DerivedRT>& RT,
-    const Eigen::VectorXi& L,
-    const Eigen::VectorXi& R,
-    const int i
+    const Eigen::Matrix<Index,-1,1>& L,
```

Comment:
Technically, `-1` should be `Eigen::Dynamic` here.

Reply:
Yes, i agree